include/irr/asset/IRenderpassIndependentPipeline.h

```diff
@@ -27,14 +28,33 @@ enum E_PRIMITIVE_TOPOLOGY
     EPT_PATCH_LIST = 10
 };
 
-struct SVertexInputParams
+enum E_VERTEX_INPUT_RATE
+{
+    EVIR_PER_VERTEX = 0,
+    EVIR_PER_INSTANCE = 1
+};
+
+struct SVertexInputAttribParams
 {
-    bool enabled;
-    bool perInstance;
     E_FORMAT format;
-    uint32_t stride;
     uint32_t offset;
 };
+struct SVertexInputBindingParams
+{
+    uint32_t binding;
+    uint32_t stride;
+    E_VERTEX_INPUT_RATE inputRate;
+};
+struct SVertexInputParams
+{
+    _IRR_STATIC_INLINE_CONSTEXPR size_t MAX_VERTEX_ATTRIB_COUNT = 16u;
+
+    uint16_t enabledAttribFlags;
+    SVertexInputAttribParams attributes[16];
+    core::smart_refctd_dynamic_array<SVertexInputBindingParams> bindings;
```

Comment:
there's no need for a malloc'ed array, just store a `uint16_t enabledBindingFlags;` next to `enabledAttribFlags`
the max binding count is 16.

Reply:
so we still want to have max vertex attrib buffer bindings limit of 16?

Reply:
theoretically they could be different (one lower, one higher) but they are usually not so.